src/Server/Program.cs

```diff
@@ -7,6 +8,8 @@
 builder.Services.AddControllersWithViews();
 builder.Services.AddRazorPages();
 
+builder.Services.AddScoped<IDataHandler, JsonDataHandler>();
```

Comment:
You may want to consider `transient` instead of `scoped` 😊 

```suggestion
builder.Services.AddTransient<IDataHandler, JsonDataHandler>();
```
More info about dependency injection lifetime [here](https://www.c-sharpcorner.com/article/understanding-addtransient-vs-addscoped-vs-addsingleton-in-asp-net-core/)